test/tests/userobjects/layered_average/block_restricted.i

```diff
@@ -34,6 +34,7 @@
     direction = x
     variable = 'u'
     block = '1'
+    bounding_block = '1'
```

Comment:
This `bounding_block` parameter seems redundant with `block` in the case of a LayeredIntegral, doesn't it?

Wouldn't a solution be to overwrite `LayeredBase::getBounds()` in `LayeredIntegral` and automatically set `bounding_block` to `block`? Or something to avoid having to specify both? I'm afraid keeping them both would be pretty confusing for a new user. Not to mention those classes are already fairly hard to understand.

Reply:
I don't think it is necessarily redundant you can envision cases where you want the LayeredIntegral executed perhaps on one block but the layers used in the LayeredIntegral defined by the bounds of a different block.  I agree in many cases block and bounding_block will be the same but I do not think that is the case in general.

Reply:
Looks like a bad decision we made before. I think `block` parameter in `LayeredBase` should be removed completely, same as `getBounds()` call in the constructor. Can we just call `getBounds()` always in `initialize()`? For `LayeredSideIntegral`, we possibly should just use `boundary` to determine the bounds instead of providing another parameter.

Reply:
Even for LayeredSideIntegral I think there a cases where the boundary on which you execute is different then the boundary used to get the bounds.  (i.e., the subdomain used to grab the bounds for the layers does not necessarily correspond to the subdomain on which the UserObject is executed). Therefore, I think you will always need another parameter.

Reply:
I did not try, but looks like we can do something like
```
auto block_restricted_pointer = dynamic_cast<LayeredIntegral *>(this);
if (block_restricted_pointer)
{
  if (block_restricted_pointer->blockRestricted())
 {
    // get bounds based on block
 }
 else
   // simply use mesh limits
}
else
{
   auto boundary_restricted_pointer = dynamic_cast<LayeredSideIntegral *>(this);
   if (!boundary_restricted_pointer)
      mooseError("How this can happen?")

   // get bounds based on boundary
}

Reply:
The problem is that if this `bounding_block` parameter is not provided, you will be forcing users to use the mesh limits instead of horning `block` or `boundary` parameter.

Reply:
@YaqiWang I'd like to discuss this PR with you today if you have time.

Reply:
Anytime feel fee to stop by.